cmd/submit.go

```diff
@@ -49,21 +49,69 @@ var submitCmd = &cobra.Command{
 }
 
 func runSubmit(cfg config.Config, flags *pflag.FlagSet, args []string) error {
-	usrCfg := cfg.UserViperConfig
+	if err := validateUserConfig(cfg.UserViperConfig); err != nil {
+		return err
+	}
+
+	ctx, err := newSubmitCmdContext(cfg.UserViperConfig, flags, args)
+	if err != nil {
+		return err
+	}
+
+	if err := ctx.submit(cfg.UserViperConfig); err != nil {
+		return err
+	}
+
+	ctx.printResult()
+	return nil
+}
+
+// submitCmdContext represents the context for the submit cmd.
+type submitCmdContext struct {
+	usrCfg *viper.Viper
+	flags  *pflag.FlagSet
+	submission
+}
+
+// newSubmitCmdContext sets up a context to initiate a submission.
+func newSubmitCmdContext(usrCfg *viper.Viper, flags *pflag.FlagSet, args []string) (*submitCmdContext, error) {
+	ctx := &submitCmdContext{usrCfg: usrCfg, flags: flags}
+
+	filepaths, err := ctx.sanitizeArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	exercise, err := ctx.exercise(filepaths)
+	if err != nil {
+		return nil, err
+	}
 
-	if usrCfg.GetString("token") == "" {
-		return fmt.Errorf(msgWelcomePleaseConfigure, config.SettingsURL(usrCfg.GetString("apibaseurl")), BinaryName)
+	if err = ctx.migrateLegacyMetadata(exercise); err != nil {
+		return nil, err
 	}
 
-	if usrCfg.GetString("workspace") == "" {
-		return fmt.Errorf(msgRerunConfigure, BinaryName)
+	documents, err := ctx._documents(filepaths, exercise)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := ctx._metadata(exercise)
+	if err != nil {
+		return nil, err
 	}
 
+	ctx.submission = submission{documents: documents, metadata: metadata}
+	return ctx, nil
+}
+
+// sanitizeArgs validates args and swaps with evaluated symlink paths.
+func (s *submitCmdContext) sanitizeArgs(args []string) ([]string, error) {
```

Comment:
The name `sanitizeArgs` is a bit confusing to me. I suspect that the function is doing two different things, both of which should be named. E.g. `ensurePathsExist` and `ensurePathsAreFiles` or something like that.